### DISCUSSION TITLE
Invalidate all routes

### DISCUSSION THREAD

**Discussion by merthanmerter** [TanStack/router#2567] [Q&A] [answered]
Is there a way to invalidate all routes?

My app is multitenant, when I change the tenant "activeTenant" in user data updates and session token refreshes. Then I navigate to root path.

```ts
const changeActiveTenant = useMutation({
    mutationFn: async (data: Pick<SelectTenant, "id">) => {
      return await context.proxy.auth.changeActiveTenant.mutate(data);
    },
    onSuccess: async () => {
      router.invalidate();
      router.navigate({ to: "/hub" });
    },
    onError: (err) => {
      toast.error(err.message);
    },
  });
```

This method only invalidates the current path that it is called, so this causes showing the old tenant data for a second before new data loads with ~~beforeLoad~~ `loader` method when navigating to a path.

---

**Edit:**
This is working perfectly fine now.
[gist link](https://gist.github.com/merthanmerter/396c850684367dcad05bb1f29c5ee721)
[repo link](https://github.com/merthanmerter/wepieces)

**Comment by schiller-manuel** [comment, marked as answer]
You are looking for a way to empty all cached matches.
we don't have this yet, but this could be added

**Reply by schiller-manuel** [reply]
@merthanmerter  I added this in https://github.com/TanStack/router/pull/2575

**Reply by merthanmerter** [reply]
> @merthanmerter  I added this in https://github.com/TanStack/router/pull/2575

I can’t find the right words to say how amazing you guys are. Thank you.
